### PR TITLE
Add Natspec indicating amount must be amount - 1 wei or less when claiming fees

### LINKED ISSUE #65: Update `claimFees` Natspec to Note that Max Amount requested must always leave 1 wei
Because the pool contracts only allow claiming up to the total amount minus 1 wei, ensure this is clearly documented in the natspec comments, and later in the unistaker docs website. See pool contract implementation here for details: https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L848-L868

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Add natspec

Closes #65

### CODE REVIEW

File: src/V3FactoryOwner.sol [hunk 1]
```diff
@@ -177,7 +177,9 @@ contract V3FactoryOwner {
   /// method.
   /// @return _amount0 The amount0 fees collected, returned by the pool's collectProtocol method.
   /// @return _amount1 The amount1 fees collected, returned by the pool's collectProtocol method.
-  /// @dev See docs on IUniswapV3PoolOwnerActions for more information on forwarded params.
+  /// @dev The `UniswapV3Pool` contract only allows claiming the total amount minus 1 wei or less
+  /// for each token. We highly recommend checking the documentation on `IUniswapV3PoolOwnerActions`
```
Comment (wildmolasses) [inline src/V3FactoryOwner.sol:181]: I don't know if we recommend the interface documentation -- is it this one? https://docs.uniswap.org/contracts/v3/reference/core/interfaces/pool/IUniswapV3PoolOwnerActions -- if so, it's not great, and makes no reference the issue at hand.

File: src/V3FactoryOwner.sol [hunk 2]
```diff
@@ -177,7 +177,9 @@ contract V3FactoryOwner {
   /// method.
   /// @return _amount0 The amount0 fees collected, returned by the pool's collectProtocol method.
   /// @return _amount1 The amount1 fees collected, returned by the pool's collectProtocol method.
-  /// @dev See docs on IUniswapV3PoolOwnerActions for more information on forwarded params.
+  /// @dev The `UniswapV3Pool` contract only allows claiming the total amount minus 1 wei or less
```
Comment (wildmolasses) [inline src/V3FactoryOwner.sol:180]: a different idea is to annotate the `@param _amount0Requested` and `@param _amount1Requested` with "max value" suggestions (`UniswapV3Pool:protocolFees.token0 - 1`, `UniswapV3Pool:protocolFees.token1 - 1`). do you think it helps at all?

File: src/V3FactoryOwner.sol (hunk 1 above)
Comment (alexkeating) [reply src/V3FactoryOwner.sol:181]: Should we direct people to the `UniswapV3Pool` contract instead? The idea with the second sentence is to encourage people to look at the underlying contract before calling the method.

File: src/V3FactoryOwner.sol (hunk 2 above)
Comment (alexkeating) [reply src/V3FactoryOwner.sol:180]: Yes, I think so. Should we keep a dev note as well or remove it altogether?

File: src/V3FactoryOwner.sol (hunk 1 above)
Comment (wildmolasses) [reply src/V3FactoryOwner.sol:181]: yeah that might be better, as the weirdness is only expressed in the contract code itself afaik

File: src/V3FactoryOwner.sol (hunk 2 above)
Comment (alexkeating) [reply src/V3FactoryOwner.sol:180]: https://github.com/uniswapfoundation/UniStaker/pull/73/commits/2b0cf560a5f07178f9b7fee6443ac68b0232c391

File: src/V3FactoryOwner.sol (hunk 1 above)
Comment (alexkeating) [reply src/V3FactoryOwner.sol]: https://github.com/uniswapfoundation/UniStaker/pull/73/commits/2b0cf560a5f07178f9b7fee6443ac68b0232c391

File: src/V3FactoryOwner.sol [hunk 3]
```diff
@@ -172,12 +172,14 @@ contract V3FactoryOwner {
...
   /// @param _amount0Requested The amount0Requested param to forward to the pool's collectProtocol
-  /// method.
+  /// method. Its maximum value will be `protocolFees.token0 - 1`.
   /// @param _amount1Requested The amount1Requested param to forward to the pool's collectProtocol
-  /// method.
+  /// method. Its maximum value will be `protocolFees.token1 - 1`.
   /// @return _amount0 The amount0 fees collected, returned by the pool's collectProtocol method.
   /// @return _amount1 The amount1 fees collected, returned by the pool's collectProtocol method.
-  /// @dev See docs on IUniswapV3PoolOwnerActions for more information on forwarded params.
+  /// @dev The `UniswapV3Pool` contract only allows claiming the total amount minus 1 wei or less
```
Comment (apbendi) [inline src/V3FactoryOwner.sol:180]: nit: Might be clearer if worded "The `UniswapV3Pool contract allows claiming a maximum of the total accrued fees minus 1"

**Review by apbendi — APPROVED** [review]
wording nit but looks good

File: src/V3FactoryOwner.sol (hunk 3 above)
Comment (alexkeating) [reply src/V3FactoryOwner.sol]: https://github.com/uniswapfoundation/UniStaker/pull/73/commits/54c4ce94ffc6ec83fd0ec76bc0b52b6c35c264b1

**Review by wildmolasses — APPROVED** [review]
lgtm

**Pull request merged**